Lustre Buildbot Changes

dinatale2 opened the pull request:
All changes incorporated into the lustre buildbot.

behlendorf committed:
Add isIdle helper

Checks if a buildslave is available and currently idle.  In the case
of a latent buildslave it may be available but not substantiated.  This
helper allows us to determine if a build can be immediately serviced
without the cost of substantiating a new buildslave.

Signed-off-by: Brian Behlendorf <behlendorf1@llnl.gov>

behlendorf committed:
Treat RETRY as FAILURE

When testing kernel modules with buildbot it's entirely possible that
the system may panic resulting in a disconnect.  For the purposes of
automated testing we want to consider this a FAILURE and immediately
move on to testing the next change.

This change should not be pushed upstream until the behavior is made
more flexible.  For example, this could be done using a 'flunkOnRetry'
build step option.

Signed-off-by: Brian Behlendorf <behlendorf1@llnl.gov>

behlendorf committed:
Set tags for latent ec2 buildslaves

EC2LatentBuildSlave does not respect the tags argument to __init__()
when launching spot instances.  Apply the tags in _wait_for_instance()
identical to what is done for attaching volumes.

http://trac.buildbot.net/ticket/2903

behlendorf committed:
Retry on EC2 'NotFound' errors

Due to EC2 eventual consistency model 'NotFound' errors can returned
even after the successful creation of a resource.  This spurious errors
should be handled by retrying the operation according to the Amazon
documentation.

http://docs.aws.amazon.com/AWSEC2/latest/APIReference/query-api-troubleshooting.html#eventual-consistency
Signed-off-by: Brian Behlendorf <behlendorf1@llnl.gov>

behlendorf committed:
Soft disconnect when 'build_wait_timeout==0'

Insubstantiating the latent build slave without cleanly disconnecting
it can result in a ping failure if a new latent slave is immediately
instantiated.  To prevent this the slave should disconnect cleanly
before being insubstantiated.

  [Broker,2,52.25.93.92] ping finished: failure
  [Broker,2,52.25.93.92] slave ping failed; re-queueing the request

This is easily reproducible by running back to back builds using EC2.

Signed-off-by: Brian Behlendorf <behlendorf1@llnl.gov>

morrone commented:
Weren't the first five or so of these patches authored by Brian?  It looks like it in the original patch set.  If you use git-am to apply the patches they will preserve authorship.  One way or another, could you fix up the authorship on the relevant commits and redo the pull request?

dinatale2 committed:
Force root volumes to be deleted upon termination on EC2 volumes.

Introduce the ability to set all root
volumes of an instance to be deleted upon termination. This change
does not impact additionally attached volumes.

Spot instances will make root volumes and are similar to on demand
instances. They will not destroy a volume unless delete on termination
is set.

Signed-off-by: Giuseppe Di Natale <dinatale2@llnl.gov>

dinatale2 committed:
Add project and branch filter to GerritChangeSource.

gerrit stream-events provides events for all projects and
branches a user has access to. Allow buildbot's GerritChangeSource
to filter changes based on branch and project.

Signed-off-by: Giuseppe Di Natale <dinatale2@llnl.gov>

dinatale2 committed:
Allow non-Linux type spot instances.

Buildbot was hardcoded to only support
Linux/Unix spot instances. Allow for a
product description to be passed into
the EC2LatentBuildSlave class.

Signed-off-by: Giuseppe Di Natale <dinatale2@llnl.gov>

dinatale2 committed:
Corrected a bug with getting property values.

ChangeFilter was calling a nonexistent method "get"
on properties. The correct method is "getProperty"

Signed-off-by: Giuseppe Di Natale <dinatale2@llnl.gov>

dinatale2 commented:
Authorship is fixed. Should be all good now.
